### ISSUE TITLE
[Feature request] Update tailwindcss ^3.4.0

### ISSUE THREAD

**Issue by you0518** [francoismassart/eslint-plugin-tailwindcss#297]
**Is your feature request related to a problem? Please describe.**
In 3.4.0 of tailwindcss, subgrid, size, etc. were added.

However, using them in the current version will result in `no-custom-classname`.

**Describe the solution you'd like**
Please update tailwindcss to 3.4.0

**Comment by francoismassart** [comment]
Hi,
I'm finally feeling better after quite an intense flu session ! (3 weeks)

I started migrating the entire plugin in TypeScript in a private repository for now.
I will publish it here, once it is decent enough 😅 but since it'll take some time to complete it, I started working on updating the plugin for `3.4.0`.

https://github.com/francoismassart/eslint-plugin-tailwindcss/tree/feat/tailwind-3.4.0

I follow the changelog from https://tailwindcss.com/blog/tailwindcss-v3-4 :

- ~[Dynamic viewport units:](https://tailwindcss.com/blog/tailwindcss-v3-4#dynamic-viewport-units)~
- ~[New :has() variant:](https://tailwindcss.com/blog/tailwindcss-v3-4#new-has-variant)~
- ~[Style children with the * variant:](https://tailwindcss.com/blog/tailwindcss-v3-4#style-children-with-the-variant)~
- **WIP** [New size-* utilities:](https://tailwindcss.com/blog/tailwindcss-v3-4#new-size-utilities)
- [Balanced headlines with text-wrap utilities:](https://tailwindcss.com/blog/tailwindcss-v3-4#balanced-headlines-with-text-wrap-utilities)
- [Subgrid support:](https://tailwindcss.com/blog/tailwindcss-v3-4#subgrid-support)
- [Extended min-width, max-width, and min-height scales:](https://tailwindcss.com/blog/tailwindcss-v3-4#extended-min-width-max-width-and-min-height-scales)
- [Extended opacity scale:](https://tailwindcss.com/blog/tailwindcss-v3-4#extended-opacity-scale)
- [Extended grid-rows-* scale:](https://tailwindcss.com/blog/tailwindcss-v3-4#extended-grid-rows-scale)
- [New forced-colors variant:](https://tailwindcss.com/blog/tailwindcss-v3-4#new-forced-colors-variant)
- [New forced-color-adjust utilities:](https://tailwindcss.com/blog/tailwindcss-v3-4#new-forced-color-adjust-utilities)

The `size-*` will be integrated with the autofix of the `enforces-shorthand` rule, merging `w-*` and `w-*` into `size-*`.

I'll publish a beta version with the support for 3.4.0 soon and I'll counting on your feedbacks.

Cheers

**Comment by francoismassart** [comment]
Here is the beta version ready for testing with Tailwind CSS 3.4.0
`npm i eslint-plugin-tailwindcss@3.14.0-beta.1`

Feedbacks are welcome 👍

**Comment by francoismassart** [comment]
I could test it out on a real project on monday. Has anyone else tried ?

`npm i eslint-plugin-tailwindcss@3.14.0-beta.1`
+
`npm i tailwindcss@3.4.0`

cc @you0518 @lassenorgaard , @tomsdob, @martinsjek, @RobertsG, @michaeltugby0, @Serator, @Yovach, @AleksandrIvanenko, @stefanofa, @Stradi ...

Thank you for your feedback

**Comment by you0518** [comment]
@francoismassart 

> npm i eslint-plugin-tailwindcss@3.14.0-beta.1
> npm i tailwindcss@3.4.0

We have confirmed that this version works. And we also confirmed that the new `tailwindcss/enforces-shorthand` appears.

```
 Warning: Classnames 'h-full, w-full' could be replaced by the 'size-full' shorthand!
```

Thank you!

**Comment by Yovach** [comment]
Hi @francoismassart, sorry I didn't see your previous message.
Here are the rules I use with your plugin:

``json
  ...
  "rules" : {
    "tailwindcss/no-custom-classname" : "warn"
  },
  ...

```

I just tested the `lint` command (without the fix), it shows me some shortcuts (`h-5` and `w-5` to `size-5`) and "invalid Tailwind CSS classnames order".

I've tried `--fix` and it seems to work.

**Comment by stefanofa** [comment]
Sorry @francoismassart, I didn't see the comment either because I didn't get the notification.

Anyway, I tested the version you mentioned on a couple of medium/large projects in production and it seems to work, even the automatic fix. The only things that were present though were `h-*` and `w-*` to `size-*`.

Thank you for your work, it is really a great value :)

**Comment by stefanofa** [comment]
I just noticed that in one place the class `min-w-[12rem]` is used, which could be replaced with `min-w-48`.
However, I do not know whether this is preferable or not.

![image](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/9802152/177f617d-8f11-44d0-83ca-b1fd9b9492ce)
![image](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/9802152/5b44d03b-da24-4282-9ad1-d30a63f292d6)

**Comment by Serator** [comment]
@francoismassart Looks like everything is working as expected. Thanks!

**Comment by francoismassart** [comment]
@stefanofa, yes! A rule autofixing the arbitrary values classnames by equivalent classnames from the config is a good idea.
I added a [note about it in the README](https://github.com/francoismassart/eslint-plugin-tailwindcss?tab=readme-ov-file#upcoming-rules)

**Comment by eramsayCB** [comment]
Hi @francoismassart, following up on this thread, I ran into a false positive warning related to responsive utilities being out of order that may not have been accounted for in the latest updates for tailwindcss@v3.4+ 

Example...

<img width="1040" alt="Screenshot 2024-03-22 at 10 48 19 AM" src="https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/88685357/ad273861-3592-4d24-9568-177b177df1ce">

Which resolves to the following using linter rules when using the latest  v3.15.1

<img width="965" alt="Screenshot 2024-03-22 at 10 48 39 AM" src="https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/88685357/2f134072-27f2-4149-bfdc-4c9f0f0a0621">

Would it be possible to update to accomodate responsive sizing for the latest utilities in tailwindcss@3.4+?
